### PR TITLE
refactor: use `expResult` in acknowledgePacket tests

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

This PR refactors a large number of tests - specifically `AcknowledgePacket` tests.

The previous code relied on exposing vars for `assertFn` as well as `expError`. This PR replaces those vars with an `expResult` function on the testcase structure.

There is no rush on this PR as it should not be affected by other changes to channel upgradability. I highly recommend reviewing using split-diffs for this PR.

closes: #4433 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/packet_test.go [hunk 1]
```diff
@@ -730,379 +730,514 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
...
+				suite.Require().Error(err)
+				suite.Require().ErrorIs(err, types.ErrPacketSequenceOutOfOrder)
+			},
+		},
 	}
 
 	for i, tc := range testCases {
 		tc := tc
-		suite.Run(fmt.Sprintf("Case %s, %d/%d tests", tc.msg, i, len(testCases)), func() {
+		suite.Run(fmt.Sprintf("Case %s, %d/%d tests", tc.name, i, len(testCases)), func() {
```
Comment (chatton) [inline modules/core/04-channel/keeper/packet_test.go:1238]: not strictly relevant to the refactor, but I think we can just pass in tc.name directly

```golang
suite.Run(tc.name, func(){})
```

The extra formatting produces extra noise in the test output that I don't think is useful. e.g. `TestAcknowledgePacket/Case_success_on_channel_in_flushing_state_with_valid_timeout,_3/20_tests`

File: modules/core/04-channel/keeper/packet_test.go [hunk 2]
```diff
@@ -730,379 +730,514 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
...
-			suite.coordinator.Setup(path)
+				// create packet commitment
+				sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+				suite.Require().NoError(err)
+				packet = types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+				channelCap = suite.chainA.GetChannelCapability(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+			},
+			expResult: func(commitment []byte, err error) {
+				suite.Require().Error(err)
+				// suite.Require().ErrorIs(err, connectiontypes.ErrConnectionNotFound)
```
Comment (chatton) [inline modules/core/04-channel/keeper/packet_test.go:1146]: looks like this case didn't assert a specific error so we can remove this?

**Review by chatton — APPROVED** [review]
Refactor looks great, much easier to reason about. One thing that might actually be a useful addition to the existing tests is a state check on the channel.

Now that this function is possibly modifying the channel state to OPEN/FLUSHCOMPLETE or leaving at FLUSHING, it might be nice to add a check for OPEN/CLOSED in all the existing tests WDYT?

File: modules/core/04-channel/keeper/packet_test.go (hunk 2 above)
Comment (damiannolan) [reply modules/core/04-channel/keeper/packet_test.go:1146]: oh my bad, I remember commenting this out and meaning to come back to it. 

I can push a commit to fix this

**Review by DimitrisJim — APPROVED** [review]
sweet

File: modules/core/04-channel/keeper/packet_test.go (hunk 2 above)
Comment (damiannolan) [reply modules/core/04-channel/keeper/packet_test.go]: Updated to correct assert `commitmenttypes.ErrInvalidProof`

**Pull request merged**